bgpd/bgp_ovsdb_if.c

```diff
@@ -2410,6 +2406,21 @@ bgp_nbr_update_source_ovsdb_apply_changes (const struct ovsrec_bgp_neighbor *ovs
     }
 }
 
+static void
+bgp_nbr_weight_ovsdb_apply_changes (const struct ovsrec_bgp_neighbor *ovs_nbr,
+    const struct ovsrec_bgp_router *ovs_bgp,
+    char *name,
+    struct bgp *bgp_instance)
+{
+    uint16_t weight;
+
+    if (COL_CHANGED (ovs_nbr, ovsrec_bgp_neighbor_col_weight, idl_seqno)) {
+        weight = (ovs_nbr->n_weight == 0 ? BGP_ATTR_DEFAULT_WEIGHT :
```

Comment:
I suggest to avoid using function `peer_weight_unset` since CLI description of command `no neighbor weight` (written by ops designers) explicitly says ""Set default weight for routes from this neighbor", thus, it should not be reset to 0.